src/diagnostics.md

```diff
@@ -0,0 +1,271 @@
+# Diagnostics
+
```

Comment:
A paragraph summarizing the contents of this page would be helpful. If not part of this PR, then please open an issue for it.